gammapy/datasets/simulate.py

```diff
@@ -169,6 +169,12 @@ def _sample_coord_time_energy(self, dataset, model):
 
         npred = self._evaluate_timevar_source(dataset, model=model)
         data = npred.data[np.isfinite(npred.data)]
+
+        if np.sum(data) > 1e6:
```

Comment:
1e6 seems a very low value. I would have assumed this would work up to 1e16 or more. Are there really issues occurring for such low values?